### ISSUE TITLE
Output to file not working in >5.4.0 

### ISSUE THREAD

**Issue by sevaru** [palantir/tslint#2863]
### Bug Report

- __TSLint version__:  5.4.2
- __TypeScript version__: 2.3.4
- __Running TSLint via__:  CLI

#### TypeScript code being linted

```ts
tslint -p tsconfig.json -c tslint.json -o tslint-output.json
```

with `tslint.json` configuration:

```json
{
    "rules": {
    	"no-var-keyword": true
    }
}
```

#### Actual behavior
With a versions start from 5.4.0 tslint-output.json file is empty. Works with 5.3.2, 5.3.0.

#### Expected behavior
tslint-output.json file has information about errors.

##### Note
I added only one rule in tslint.json just for sample purpose. It doesn't work with any other rule too.

**Comment by ajafff** [comment]
Seems like a race condition between asynchronously writing to file and the call to `process.exit(code)` after linting.
PR incoming.
